runtime/compiler/control/J9Options.cpp

```diff
@@ -1599,6 +1602,26 @@ void J9::Options::preProcessJniAccelerator(J9JavaVM *vm)
       }
    }
 
+uint64_t getCodeCacheMaxPercentageOfAvailableMemory(J9JavaVM *vm)
+   {
+   PORT_ACCESS_FROM_JAVAVM(vm);
+   OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+   uint64_t codeCacheTotalPercentage = CODECACHE_DEFAULT_RAMPERCENTAGE;
+   char *xxccPercentOption = J9::Options::_externalOptionStrings[J9::ExternalOptions::XXcodecachetotalRAMPercentage];
+   int32_t XXcodeCacheTotalPercentArg = FIND_ARG_IN_VMARGS(EXACT_MEMORY_MATCH, xxccPercentOption, 0);
+   if (XXcodeCacheTotalPercentArg >= 0)
+      {
+      IDATA returnCode = GET_MEMORY_VALUE(XXcodeCacheTotalPercentArg, xxccPercentOption, codeCacheTotalPercentage);
+      if (OPTION_OK == returnCode)
+         {
+         if ((codeCacheTotalPercentage < 0.0) || (codeCacheTotalPercentage > 100.0))
+            j9nls_printf(PORTLIB, J9NLS_WARNING, J9NLS_JIT_OPTIONS_INCORRECT_MEMORY_SIZE, codeCacheTotalPercentage);
```

Comment:
I wonder if this needs to be a more precise message

Reply:
Currently the error we get for out of range is `JVMJITM031W Value for '-XX:codecachetotalMaxRAMPercentage=' is not correct` . I could not find anything in the existing nls file with better error message. If you think we should add a new message then I can try something like: `J9NLS_JIT_OPTIONS_PERCENT_OUT_OF_RANGE=Value for '%s' must be between 0.0 and 100.0`